internal/store/postgres/scheduler/replay_repository.go

```diff
@@ -343,21 +328,13 @@ func (r ReplayRepository) updateReplayRequest(ctx context.Context, id uuid.UUID,
 }
 
 func (r ReplayRepository) updateReplayRuns(ctx context.Context, id uuid.UUID, runs []*scheduler.JobRunStatus) error {
-	tx, err := r.db.BeginTx(ctx, pgx.TxOptions{})
-	if err != nil {
-		return err
-	}
-
-	deleteRuns := `DELETE FROM replay_run WHERE replay_id = $1`
-	if _, err := tx.Exec(ctx, deleteRuns, id); err != nil {
-		tx.Rollback(ctx)
-		return errors.Wrap(scheduler.EntityJobRun, "unable to delete runs of replay", err)
-	}
-	if err := r.insertReplayRuns(ctx, tx, id, runs); err != nil {
-		tx.Rollback(ctx)
-		return errors.Wrap(scheduler.EntityJobRun, "unable to insert runs of replay", err)
+	query := `UPDATE replay_run SET status=$1, updated_at=NOW() WHERE replay_id=$2 AND scheduled_at=$3 AND status<>$1`
+	for _, run := range runs {
+		_, err := r.db.Exec(ctx, query, run.State, id, run.ScheduledAt)
```

Comment:
What if no rows are updated in the query as we are doing a status check ?
We might need to check if the RowsAffected is 0.

Reply:
What kind of check we need to do if no rows affected @sbchaos ? If no rows updated then it means all runs between optimus and airflow are in sync. Which means the run status on airflow are not yet changed